src/btree/bt_delete.c

```diff
@@ -185,16 +186,27 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
     }
 
     /*
-     * If the page is still "deleted", it's as we left it, simply reset the state. Otherwise, the
-     * page is in an in-memory state, which means it was instantiated at some point. Walk any list
-     * of update structures and abort them. We can't use the normal read path to get the pages with
-     * updates (the original page may have split, so there many be more than one page), because the
-     * session may have closed the cursor, we no longer have the reference to the tree required for
-     * a hazard pointer. We're safe since pages with unresolved transactions aren't going anywhere.
+     * There are two possible cases:
+     *
+     * 1. The state is WT_REF_DELETED. In this case ft_info.del cannot be null, because the
+     * operation cannot reach global visibility while its transaction remains uncommitted. The page
+     * itself is as we left it, so we can just reset the state.
+     *
+     * 2. The state is WT_REF_MEM. In this case the tree must be read-write (since readonly trees
+     * can't have uncommitted write operations), and it's been instantiated, so we check
+     * ft_info.update for a list of updates to abort. Allow the update list to be null to be
+     * conservative.
      */
     if (current_state == WT_REF_DELETED)
         current_state = ref->ft_info.del->previous_ref_state;
     else if ((updp = ref->ft_info.update) != NULL)
+        /*
+         * Walk any list of update structures and abort them. We can't use the normal read path to
+         * get the pages with updates (the original page may have split, so there many be more than
```

Comment:
many -> may

Reply:
Fixed, thanks (in [2e5af24](https://github.com/wiredtiger/wiredtiger/pull/7909/commits/2e5af24e34f8557ee0cb39ab8a1f28f054c35c84))